src/Processors/Formats/Impl/JSONEachRowRowInputFormat.cpp

```diff
@@ -186,6 +187,11 @@ void JSONEachRowRowInputFormat::readJSONObject(MutableColumns & columns)
             JSONUtils::skipColon(*in);
             readField(column_index, columns);
         }
+        if (seen_columns_count >= total_columns)
+        {
+            skipToUnescapedNextLineOrEOF(*in);
```

Comment:
Also string fields can have `\n`, and it will break this logic:
```sql
select * from format(JSONEachRow, 'a UInt32', '{"a" : 42, "b" : "\n"}, {"a" : 43}')
```
So yes, we need to skip up to the end of JSON object

Reply:
I will create a PR with fix for `syncAfterError` with proper data skipping